File: src/test/java/com/thebluehats/server/game/commands/GameCommandTest.java
package com.thebluehats.server.game.commands;

import static org.junit.Assert.assertEquals;

import org.bukkit.ChatColor;
import org.bukkit.entity.Player;
import org.junit.Test;

public class GameCommandTest {
    @Test
    public void FormatsTheUsageMessageCorrectly() {
        GameCommand gameCommand = new GameCommand() {
            public String[] getCommandNames() {
                return null;
            }

            public String getUsageMessage(String cmd) {
                return "";
            }

            public void runCommand(Player player, String commandName, String[] args) {
            }
        };

        String formattedUsageMessage = gameCommand.formatStandardUsageMessage("give", "gives an item", "player", "item",
                "amount");

        String expectedMessage = ChatColor.DARK_PURPLE + "/give - " + ChatColor.RED + "gives an item"
                + ChatColor.DARK_PURPLE + " | Usage" + ChatColor.DARK_PURPLE + "/give " + ChatColor.RED
                + "<player> <item> <amount>";

        assertEquals(expectedMessage, formattedUsageMessage);
    }

    @Test
    public void FormatsTheErrorMessageCorrectly() {
        GameCommand gameCommand = new GameCommand() {
            public String[] getCommandNames() {
                return null;
            }

            public String getUsageMessage(String cmd) {
                return "";
            }

            public void runCommand(Player player, String commandName, String[] args) {
            }
        };

        String formattedErrorMessage = gameCommand.formatStandardErrorMessage("Something happend!");

        String expectedMessage = ChatColor.DARK_PURPLE + "Error! " + ChatColor.RED + "Something happend!";

        assertEquals(expectedMessage, formattedErrorMessage);
    }
}

File: minecraft-server/src/test/kotlin/com/thebluehats/server/game/commands/GameCommandTest.kt
package com.thebluehats.server.game.commands

import io.kotest.core.spec.style.DescribeSpec
import io.mockk.mockk
import io.mockk.spyk
import io.mockk.verify
import org.bukkit.entity.Player

class GameCommandTest : DescribeSpec({
    describe("game command") {
        it("executes a command based on an alias") {
            val gc: GameCommand = spyk(object : GameCommand() {
                override val commandNames: Array<String> get() = arrayOf("a", "b", "c", "d")

                override fun getUsageMessage(cmd: String?): String? = null

                override fun runCommand(player: Player, cmd: String?, args: Array<String>) {}
            })

            gc.onCommand(mockk<Player>(), mockk(), "b", emptyArray())

            verify { gc.runCommand(any(), any(), any()) }
        }
    }
})
